firewall.py

```diff
@@ -0,0 +1,97 @@
+        """
+        self.host = host
+        self.logger = logging.getLogger("firewall")
+
+    def enable(self):
+        self.logger.info("Enabling and configuring firewall...")
+        self.host.run("systemctl enable firewalld")
+        self.host.run("systemctl start firewalld")
+        self.host.run("firewall-cmd --set-default-zone=drop")
+        self.host.run("firewall-cmd --permanent --new-zone=internal || true")
+
+        # Trusted CIDRs (add as source and rich-rule to internal zone)
+        for cidr in self.TRUSTED_CIDRS:
+            self.host.run(f"firewall-cmd --permanent --zone=internal --add-source={cidr}")
+            self.host.run(f"firewall-cmd --permanent --zone=internal --add-rich-rule=rule family=\'ipv4\' destination address=\'{cidr}\' accept")
+
+        # API Server IP (add as source and rich-rule to internal zone)
+        #api_ip = self.API_SERVER_IP
+        #self.host.run(f"firewall-cmd --permanent --zone=internal --add-source={api_ip}/32")
+        #self.host.run(f"firewall-cmd --permanent --zone=internal --add-rich-rule=rule family=\'ipv4\' destination address=\'{api_ip}\' accept")
+
+        # Port rules
+        for (zone, proto), ports in self.PORT_RULES.items():
+            for port in ports:
+                self.host.run(f"firewall-cmd --permanent --zone={zone} --add-port={port}/{proto}")
+
+        # SSH interfaces (add all to public zone)
+        for iface in self.SSH_INTERFACES:
+            self.host.run(f"firewall-cmd --permanent --zone=public --add-interface={iface}")
+
+        # Always open 6443/tcp on drop zone
+        self.host.run("firewall-cmd --permanent --zone=drop --add-port=6443/tcp")
+
+        # Remove mDNS from public zone and set log-denied=all
+        self.host.run("firewall-cmd --permanent --zone=public --remove-service=mdns")
+        self.host.run("firewall-cmd --set-log-denied=all")
+
+        self.host.run("firewall-cmd --reload")
+        self.logger.info("Firewall enabled and configured.")
+
+    def disable(self):
+        self.logger.info("Disabling firewall rules and cleaning up...")
+        self.host.run("systemctl enable firewalld")
+        self.host.run("systemctl start firewalld")
+        self.host.run("firewall-cmd --set-default-zone=public")
+
+        # Trusted CIDRs (remove as source and rich-rule from internal zone)
+        for cidr in self.TRUSTED_CIDRS:
+            self.host.run(f"firewall-cmd --permanent --zone=internal --remove-source={cidr}")
+            self.host.run(f"firewall-cmd --permanent --zone=internal --remove-rich-rule=rule family=\'ipv4\' destination address=\'{cidr}\' accept")
+
+        # API Server IP (remove as source and rich-rule from internal zone)
+        #api_ip = self.API_SERVER_IP
+        #self.host.run(f"firewall-cmd --permanent --zone=internal --remove-source={api_ip}/32")
+        #self.host.run(f"firewall-cmd --permanent --zone=internal --remove-rich-rule=rule family=\'ipv4\' destination address=\'{api_ip}\' accept")
```

Comment:
what is the significance of this code comment? I think it should be dropped, or the comment should make it clear why it is there and what to make of it.